### ISSUE TITLE
Image Classification using Convolutional Neural Networks

### ISSUE THREAD

**Issue by UTSAVS26** [abhisheks008/DL-Simplified#730]
### ML-Crate Repository (Proposing new issue)

🔴 **[Project Addition]: Image Classification using Convolutional Neural Networks (CNN)**

🔴 **Description:** Create a CNN-based model for image classification. You could use a popular dataset like CIFAR-10 or MNIST. Develop a web interface to upload images and get classification results.

🔴 **Dataset Link:** [CIFAR-10 Dataset](https://www.cs.toronto.edu/~kriz/cifar.html) or [MNIST Dataset](http://yann.lecun.com/exdb/mnist/)

🔴 **Approach:** 
- Implement a Convolutional Neural Network (CNN) model for image classification using a popular dataset (CIFAR-10 or MNIST).
- Use data preprocessing and augmentation techniques to enhance the model's performance.
- Compare the CNN model with at least one other image classification algorithm to highlight its effectiveness.
- Develop a web interface where users can upload images and receive classification results.
- Perform thorough exploratory data analysis (EDA) before model creation to understand the dataset's characteristics.

======================================================================================
📍 **Follow the Guidelines to Contribute to the Project:**
You need to create a separate folder named as the Project Title.
Inside that folder, there will be four main components:
- **Images:** To store the required images.
- **Dataset:** To store the dataset or information/source about the dataset.
- **Model:** To store the machine learning model you've created using the dataset.
- **requirements.txt:** This file will contain the required packages/libraries to run the project on other machines.
Inside the **Model** folder, the README.md file must be filled up properly, with proper visualizations and conclusions.

======================================================================================
🔴🟡 **Points to Note:**
- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include the issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before starting to contribute.

======================================================================================
✅ **To be Mentioned while taking the issue:**
- **Full Name:** Utsav Singhal
- **GitHub Profile Link:** [https://github.com/UTSAVS26](https://github.com/UTSAVS26)
- **Email ID:** utsavsinghal26@gmail.com
- **Participant ID (If not, then put NA):** Contributor
- **Approach for this Project:**
  - Implement a CNN model for image classification using the CIFAR-10 or MNIST dataset.
  - Perform data preprocessing and augmentation.
  - Compare the CNN model with another classification algorithm.
  - Develop a web interface for image upload and classification results.
  - Conduct EDA to understand the dataset.

- **What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.):** SSOC

======================================================================================
Happy Contributing 🚀

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
What are the 3-4 models you are planning to implement here?
@UTSAVS26

**Comment by UTSAVS26** [comment]
Hi @abhisheks008 thank you for assigning this issue to me. 
I am planning to implement these models right now:
1. **Simple CNN Model**
2. **VGG-16 Model**
3. **ResNet-50 Model**
4. **LeNet-5 Model**
5. **InceptionV3 Model**
6. **DenseNet Model**
7. **MobileNet Model**
8. **Comparison Model (e.g., Random Forest or SVM with HOG features)**
9. **AlexNet Model**
10. **EfficientNet Model**

And if any models failed based on the preprocessing and EDA I do then I will leave that model right now and work on that model later on.

**Comment by abhisheks008** [comment]
> Hi @abhisheks008 thank you for assigning this issue to me. I am planning to implement these models right now:
> 
> 1. **Simple CNN Model**
> 2. **VGG-16 Model**
> 3. **ResNet-50 Model**
> 4. **LeNet-5 Model**
> 5. **InceptionV3 Model**
> 6. **DenseNet Model**
> 7. **MobileNet Model**
> 8. **Comparison Model (e.g., Random Forest or SVM with HOG features)**
> 9. **AlexNet Model**
> 10. **EfficientNet Model**
> 
> And if any models failed based on the preprocessing and EDA I do then I will leave that model right now and work on that model later on.

Go ahead with the models.
